Src/Base/AMReX_Box.H

```diff
@@ -1565,225 +1653,323 @@ Box adjCell (const Box& b, Orientation face, int len=1) noexcept
-            0};
-#elif (AMREX_SPACEDIM == 3)
-    return {amrex::max(b1.smallend[0], lo.x),
-            amrex::max(b1.smallend[1], lo.y),
-            amrex::max(b1.smallend[2], lo.z)};
-#endif
+    return max(b1.smallEnd(), IntVectND<dim>(lo)).dim3();
 }
 
 // Min of upper bound
+template<int dim, std::enable_if_t<( 1<=dim && dim<=3 ), int> = 0>
 [[nodiscard]]
 AMREX_GPU_HOST_DEVICE
 AMREX_FORCE_INLINE
-Dim3 min_ubound (Box const& b1, Box const& b2) noexcept
+Dim3 min_ubound (BoxND<dim> const& b1, BoxND<dim> const& b2) noexcept
 {
-#if (AMREX_SPACEDIM == 1)
-    return {amrex::min(b1.bigend[0], b2.bigend[0]),
-            0, 0};
-#elif (AMREX_SPACEDIM == 2)
-    return {amrex::min(b1.bigend[0], b2.bigend[0]),
-            amrex::min(b1.bigend[1], b2.bigend[1]),
-            0};
-#elif (AMREX_SPACEDIM == 3)
-    return {amrex::min(b1.bigend[0], b2.bigend[0]),
-            amrex::min(b1.bigend[1], b2.bigend[1]),
-            amrex::min(b1.bigend[2], b2.bigend[2])};
-#endif
+    return min(b1.bigEnd(), b2.bigEnd()).dim3();
 }
 
+template<int dim, std::enable_if_t<( 1<=dim && dim<=3 ), int> = 0>
 [[nodiscard]]
 AMREX_GPU_HOST_DEVICE
 AMREX_FORCE_INLINE
-Dim3 min_ubound (Box const& b1, Dim3 const& hi) noexcept
+Dim3 min_ubound (BoxND<dim> const& b1, Dim3 const& hi) noexcept
 {
-#if (AMREX_SPACEDIM == 1)
-    return {amrex::min(b1.bigend[0], hi.x),
-            0, 0};
-#elif (AMREX_SPACEDIM == 2)
-    return {amrex::min(b1.bigend[0], hi.x),
-            amrex::min(b1.bigend[1], hi.y),
-            0};
-#elif (AMREX_SPACEDIM == 3)
-    return {amrex::min(b1.bigend[0], hi.x),
-            amrex::min(b1.bigend[1], hi.y),
-            amrex::min(b1.bigend[2], hi.z)};
-#endif
+    return min(b1.bigEnd(), IntVectND<dim>(hi)).dim3();
 }
 
+template<int dim>
 [[nodiscard]]
 AMREX_GPU_HOST_DEVICE
 AMREX_FORCE_INLINE
-Box minBox (Box const& b1, Box const& b2, IndexType typ) noexcept
-{
-#if (AMREX_SPACEDIM == 1)
-    return Box(IntVect(amrex::max(b1.smallend[0], b2.smallend[0])),
-               IntVect(amrex::min(b1.bigend  [0], b2.bigend  [0])),
-               typ);
-#elif (AMREX_SPACEDIM == 2)
-    return Box(IntVect(amrex::max(b1.smallend[0], b2.smallend[0]),
-                       amrex::max(b1.smallend[1], b2.smallend[1])),
-               IntVect(amrex::min(b1.bigend  [0], b2.bigend  [0]),
-                       amrex::min(b1.bigend  [1], b2.bigend  [1])),
-               typ);
-#elif (AMREX_SPACEDIM == 3)
-    return Box(IntVect(amrex::max(b1.smallend[0], b2.smallend[0]),
-                       amrex::max(b1.smallend[1], b2.smallend[1]),
-                       amrex::max(b1.smallend[2], b2.smallend[2])),
-               IntVect(amrex::min(b1.bigend  [0], b2.bigend  [0]),
-                       amrex::min(b1.bigend  [1], b2.bigend  [1]),
-                       amrex::min(b1.bigend  [2], b2.bigend  [2])),
-               typ);
-#endif
+BoxND<dim> minBox (BoxND<dim> const& b1, BoxND<dim> const& b2, IndexTypeND<dim> typ) noexcept
```

Comment:
Let's remove this function. I don't think it's being used anywhere now. The name of this function is very misleading. So to avoid confusion, let's just remove it.

Reply:
All minBox functions or just that one?

Reply:
Just this one. This one has different behavior from others. It was added by me 5 years ago. I made a mistake in naming.